.idea/checkstyle-idea.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CheckStyle-IDEA">
+    <option name="configuration">
+      <map>
+        <entry key="checkstyle-version" value="8.28" />
+        <entry key="copy-libs" value="false" />
+        <entry key="location-0" value="BUNDLED:(bundled):Sun Checks" />
+        <entry key="location-1" value="BUNDLED:(bundled):Google Checks" />
+        <entry key="scan-before-checkin" value="false" />
+        <entry key="scanscope" value="JavaOnly" />
```

Comment:
I think this should be "JavaOnlyWithTests"..

Reply:
IIRC, our POM is configured to ignore tests at the moment. I think we can/should be more strict, but that's not what we're starting from. Maybe it's worth discussing with @busbey and @HorizonNet ; I think those two have done the most work with Checkstyle, as far as I am aware.

Given we don't error out the build on violations, I think it would be fine to cover the test code.

Reply:
We are less strict with some rules when it comes to tests, but they are still scanned by Checkstyle. With that it should be `JavaOnlyWithTests`.

Reply:
> IRC, our POM is configured to ignore tests at the moment. 

Not totally sure about this. Here [1] is one instance where a test file is flagged by the checkstyle bot.
[1] https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1039/2/artifact/out/diff-checkstyle-hbase-server.txt